compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELowerToAIE.cpp

```diff
@@ -436,16 +484,28 @@ LogicalResult npuDmaCpyNdOpToAIE(IRRewriter &rewriter,
         empty, empty, staticOffsets, staticSizes, staticStrides,
         objFifo.getName(), 0, issueToken);
   }
-  if (dmaOp.hasTargetAddressing()) {
+  if (dmaOp.hasTargetAddressing() || dmaOp.getTargetMemorySpaceAsUInt() == 0) {
+    // DmaOp has target addressing on L3 either explicitly or implicitly.
```

Comment:
You're not really ensuring here that the addressing is on L3 side. This condition would still succeed if `dmaOp.hasTargetAddressing() == true` and `dmaOp.getTargetMemorySpaceAsUInt() != 0`, so it might be good to reflect that in the comment.